tests/trace/test_custom_objs.py

```diff
@@ -14,3 +33,43 @@ def test_decode_custom_obj_known_type(client):
 
     assert isinstance(decoded, Image.Image)
     assert decoded.tobytes() == img.tobytes()
+
+
+def test_inline_custom_obj(client):
+    dt = datetime(2025, 3, 7, 0, 0, 0)
+    encoded = encode_custom_obj(dt)
```

Comment:
Why is this `encode_custom_obj` but below it's `decode_inline_obj`?  Is there no encode inline path?

Reply:
encode_custom_obj handles both inline and file-based. I agree the asymmetry is kind of ick. I think `from_json` should not be digging into the internals of custom obj decoding as much as it is. I was a little wary of making too extreme a change to this delicate and important part of the code, but I can take a crack at it if you would like.

Reply:
If it's not too hard I would try to do that now, or as a fast follow.  At minimum we should document this in the serialization doc

Reply:
The longer I looked the harder it seemed. I updated some method names for clarity and I'm adding notes to the README.